CombinatorialGames/IGame/IGame.lean

```diff
@@ -0,0 +1,529 @@
+  refine lift (fun x ↦ mk '' range x.moveRight) fun x y h ↦ ?_
+  ext z
+  simp_rw [mem_image, mem_range, exists_exists_eq_and]
+  constructor <;> rintro ⟨i, rfl⟩
+  · obtain ⟨j, hj⟩ := h.moveRight i
+    exact ⟨j, hj.mk_eq.symm⟩
+  · obtain ⟨j, hj⟩ := h.moveRight_symm i
+    exact ⟨j, hj.mk_eq⟩
+
+@[simp] theorem leftMoves_mk (x : PGame) : leftMoves (mk x) = mk '' range x.moveLeft := rfl
+@[simp] theorem rightMoves_mk (x : PGame) : rightMoves (mk x) = mk '' range x.moveRight := rfl
+
+instance (x : IGame.{u}) : Small.{u} x.leftMoves := by
+  cases x
+  rw [leftMoves_mk]
+  infer_instance
+
+instance (x : IGame.{u}) : Small.{u} x.rightMoves := by
+  cases x
+  rw [rightMoves_mk]
+  infer_instance
+
+@[ext]
+theorem ext {x y : IGame} (hl : x.leftMoves = y.leftMoves) (hr : x.rightMoves = y.rightMoves) :
+    x = y := by
+  cases x with | H x =>
+  cases y with | H y =>
+  dsimp at hl hr
+  refine (identical_iff.2 ⟨⟨?_, ?_⟩, ⟨?_, ?_⟩⟩).mk_eq <;> intro i
+  · obtain ⟨_, ⟨j, rfl⟩, hj⟩ := hl ▸ mem_image_of_mem mk (mem_range_self (f := x.moveLeft) i)
+    exact ⟨j, mk_eq_mk.1 hj.symm⟩
+  · obtain ⟨_, ⟨j, rfl⟩, hj⟩ := hl ▸ mem_image_of_mem mk (mem_range_self (f := y.moveLeft) i)
+    exact ⟨j, mk_eq_mk.1 hj⟩
+  · obtain ⟨_, ⟨j, rfl⟩, hj⟩ := hr ▸ mem_image_of_mem mk (mem_range_self (f := x.moveRight) i)
+    exact ⟨j, mk_eq_mk.1 hj.symm⟩
+  · obtain ⟨_, ⟨j, rfl⟩, hj⟩ := hr ▸ mem_image_of_mem mk (mem_range_self (f := y.moveRight) i)
+    exact ⟨j, mk_eq_mk.1 hj⟩
+
+/-- `IsOption x y` means that `x` is either a left or a right move for `y`. -/
+def IsOption (x y : IGame) : Prop :=
+  x ∈ y.leftMoves ∪ y.rightMoves
+
+theorem IsOption.of_mem_leftMoves {x y : IGame} : x ∈ y.leftMoves → IsOption x y := .inl
+theorem IsOption.of_mem_rightMoves {x y : IGame} : x ∈ y.rightMoves → IsOption x y := .inr
+
+-- TODO: is there some more general theorem about well-founded relations on quotients
+-- that we could use here?
+theorem isOption_wf : WellFounded IsOption := by
+  suffices ∀ x, Acc IsOption (mk x) from ⟨ind this⟩
+  intro x
+  induction x using PGame.moveRecOn with
+  | IH x hl hr =>
+    constructor
+    rintro ⟨y⟩ (h | h) <;>
+    obtain ⟨_, ⟨i, rfl⟩, (hi : _ = Quot.mk _ _)⟩ := h
+    exacts [hi ▸ hl i, hi ▸ hr i]
+
+instance : IsWellFounded _ IsOption := ⟨isOption_wf⟩
+
+/-- **Conway recursion**: build data for a game by recursively building it on its
+left and right sets.
+
+See `ofSetsRecOn` for an alternate form. -/
+@[elab_as_elim]
+def moveRecOn {P : IGame → Sort*} (x)
+    (H : Π x, (Π y ∈ x.leftMoves, P y) → (Π y ∈ x.rightMoves, P y) → P x) : P x :=
+  isOption_wf.recursion x fun x IH ↦
+    H x (fun _ h ↦ IH _ (.of_mem_leftMoves h)) (fun _ h ↦ IH _ (.of_mem_rightMoves h))
+
+@[simp]
+theorem moveRecOn_eq {P : IGame → Sort*} (x)
+    (H : Π x, (Π y ∈ x.leftMoves, P y) → (Π y ∈ x.rightMoves, P y) → P x) :
+    moveRecOn x H = H x (fun y _ ↦ moveRecOn y H) (fun y _ ↦ moveRecOn y H) :=
+  isOption_wf.fix_eq ..
+
+-- TODO: docstring
```

Comment:
This should be 'proper subposition': a 'subposition' would be defined over `Relation.ReflTransGen IsOption`.
I'm unaware of any use of Subsequent.

Reply:
Making this reflexive sort of defeats its purpose, since it becomes no longer well-founded. Otherwise I agree with renaming this to `Subposition`.